### DISCUSSION TITLE
Insert a new record in a second table when I edit a record in a first table

### DISCUSSION THREAD

**Discussion by LucaCVG** [filamentphp/filament#10236] [Help] [answered]
Hi everyone! Using the record edit form, I would like to be able to save some data from this form (EditWbe of WbeResource) in a second table (of EarnedValueResource), creating a new record in the second table every time I modify the record in the first.

However without success. There are no errors on the screen, but no record is created in the second table, as if the code written below was not executed.

Inside "EditWbe.php" (of "WbeResource") I inserted this portion of code:

`

namespace App\Filament\Resources\WbeResource\Pages;

use App\Filament\Resources\WbeResource;
use Filament\Actions;
use Filament\Resources\Pages\EditRecord;
use Filament\Resources\Pages\CreateRecord;
use Illuminate\Database\Eloquent\Model;
use App\Models\EarnedValue;

class CreateEarnedValueDetection extends CreateRecord
{
    protected static string $resource = EarnedValueResource::class;

    protected function handleRecordUpdate(Model $record, array $data): Model
    {
        
        $record =  static::getModel()::create($data);

        // Insert values into array
        $EVdetection = new EarnedValue();
        $EVdetection->name = $data['name'];
        $EVdetection->planned_value = $data['planned_value'];
        $EVdetection->earned_value = $data['earned_value'];
        $EVdetection->schedule_variance = $data['schedule_variance'];
        $EVdetection->cost_variance = $data['cost_variance'];
        $EVdetection->schedule_perfomance_index = $data['schedule_perfomance_index'];
        $EVdetection->cost_performance_index = $data['cost_performance_index'];
        $EVdetection->estimated_to_completation = $data['estimated_to_completation'];
        $EVdetection->estimate_to_complete = $data['estimate_to_complete'];
        
        // Insert FK into array 
        $EVdetection->practice_id = $record->practice_id; 
        $EVdetection->practice_manager_id = $record->practice_manager_id; 
        $EVdetection->project_manager_id = $record->project_manager_id; 
        $EVdetection->wbe_id = $record->wbe_id; 

        // Save the Earned Value Detection model to insert the data
        $EVdetection->save();

        return $record;
    }
}

`

in "Wbe.php" model:

`
    public function earnedvalue(): HasMany
    {
        return $this->hasMany(EarnedValue::class , 'id');
    }
`

in "EarnedValue.php" model:

`
    public function wbe(): BelongsTo
    {
        return $this->belongsTo(Wbe::class , 'wbe_id');
    }
`

"EarnedValueResource" (Resource and Page) exists, but is not used because no records here will be inserted manually by the FrontEnd.

I need this because in the second table I have to keep track of all the times the record is modified to build a graph that shows me the change in values ​​over time.

**Comment by kombiahrk** [comment, marked as answer]
**Use this method in Create Page**
protected function afterCreate(): void
    {
        $record = $this->record;
//insert your coding here
     }
     
 **Use this method in Edit Page**
protected function afterSave(): void
    {
        $record = $this->record;
//insert your coding here
     }

**Reply by LucaCVG** [reply]
it worked! Thank you!!!

**Reply by LucasCis92** [reply]
Hi! Would you explain this code in more depth if you could? I'm dealing with the same issue right now, thank you!
